### PR TITLE
Add a PatientSchema type

### PR DESCRIPTION
**Pull request by clemens-msupply**
Part of #1840

### CODE REVIEW

File: client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx [hunk 1]
```diff
@@ -154,7 +154,10 @@ const queryMatchTypes = {
   dateOfBirth: 'equalTo',
 };
 
-const createSearchFilter = (searchFields: string[], data: any) => {
+const createSearchFilter = (
+  searchFields: string[],
+  data: Record<string, string | undefined> | undefined
```
Comment (clemens-msupply) [inline client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx:159]: unrelated, just fixing a warning...

**Comment by CarlosNZ** [comment]
I'll have a look at this once #2216 is locked off -- I think it's pretty good now barring things that have been moved to new issues.

**Comment by CarlosNZ** [comment]
Just need to update from `develop` and fix conflicts :)

**Review by CarlosNZ — COMMENTED** [review]
This seems fine, but I'm a bit unsure why it's necessary. You've changed the type from PatientRowFragment to the new PatientSchema, but we never actually do anything downstream with any "Patient" objects that would be affected by this. It's only at runtime that the schema of the patient matters, and that's just in JSON and we can't do anything about that anyway.

I get that it would be preferable for the compiler if the code was doing something with specific fields on Patient, but it's not -- we can take any kind of Patient as long as it has those common "PatientFragment" fields. Whose to say a patient needs to comply with this schema? It could be anything (as long as it overlaps with the basic patient type), which is why we've got in in JSON schema documents rather than hard-coding it into the app. So why add this hard-coded schema in here now?

**Comment by clemens-msupply** [comment]
> So why add this hard-coded schema in here now?

The added base patient schema is actually enforced by the backend, i.e. you can only update a patient if it fulfils the patient base schema. So at the input boundary we can define what we expect and from there on we are type safe.

Its also not directly obvious (at least not to me) that we don't use anything from the results. I also would like to avoid re-evaluating this when the code get changed. Found there are surprisingly many problems when you let `any` slip into your code.

Alternatively we might have been able to use `unknown` instead of `any` or just define the minimal required overlap. But I think the patient schema might come in handy elsewhere as well...

**Comment by CarlosNZ** [comment]
> Its also not directly obvious (at least not to me) that we don't use anything from the results.

We shouldn't be using any fields in the code base that are not part of the core data types, as we can't rely on them being there in other schemas. Someone else might use a different Patient schema, which would be inconsistent with this as it's only relevant for PNG.

I just don't understand why we'd want to hard-code one particular version of the schema when we are in no way requiring that schema to be used for Patient documents elsewhere -- that's the whole point of the JSON Forms isn't it -- the schema is changed depending on the specific use case.

Maybe you can explain to me on Monday if you think it's necessary. I guess I'm just not seeing the problem that this supposedly solves (or maybe there's a better way?)

**Comment by clemens-msupply** [comment]
The schema in this PR is not the PNG schema, it is the common base schema. It's the same for all patient types.

**Review by CarlosNZ — APPROVED** [review]
Okay, that's fine, thanks.

Only suggestion I would make would be maybe change the type name to `BasePatientSchema` or something similar to make it clear that it's not *the* definitive patient schema.

**Pull request merged**